### PR TITLE
Add Profile Readme for Organisations

### PR DESCRIPTION
**Pull request by 6543**
https://blog.gitea.com/release-of-1.20.0/#-user-profile-readme-23260 (#23260) did introduce Profile Readme for Users.

This makes it usable for Organisations:

image

---
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

File: routers/web/org/home.go [hunk 1]
```diff
@@ -155,5 +157,31 @@ func Home(ctx *context.Context) {
...
 	ctx.Data["ShowMemberAndTeamTab"] = ctx.Org.IsMember || len(members) > 0
 
+	profileGitRepo, profileReadmeBlob, profileClose := shared_user.FindUserProfileReadme(ctx)
+	defer profileClose()
+	calcOrgProfileReadme(ctx, profileGitRepo, profileReadmeBlob)
+
 	ctx.HTML(http.StatusOK, tplOrgHome)
 }
+
+func calcOrgProfileReadme(ctx *context.Context, profileGitRepo *git.Repository, profileReadme *git.Blob) {
```
Comment (KN4CK3R) [inline routers/web/org/home.go]: `calc`? `prepare` or similar would be better.

File: routers/web/org/home.go [hunk 2]
```diff
@@ -155,5 +157,31 @@ func Home(ctx *context.Context) {
...
+	if profileGitRepo == nil || profileReadme == nil {
+		ctx.Data["HasProfileReadme"] = false
```
Comment (KN4CK3R) [inline routers/web/org/home.go]: This variable is not be necessary. Just use `{{if .ProfileReadme}}` in the template.

File: templates/org/home.tmpl [hunk 3]
```diff
@@ -38,6 +38,10 @@
 	<div class="ui container">
 		<div class="ui mobile reversed stackable grid">
 			<div class="ui {{if .ShowMemberAndTeamTab}}eleven wide{{end}} column">
+				{{if .HasProfileReadme}}
```
Comment (KN4CK3R) [inline templates/org/home.tmpl]:
```suggestion
				{{if .ProfileReadme}}
```

File: docs/content/usage/profile-readme.en-us.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ menu:
 
 # Profile READMEs
 
-To display a Markdown file in your Gitea profile page, simply create a repository named `.profile` and add a new file called `README.md`. Gitea will automatically display the contents of the file on your profile, above your repositories.
+To display a Markdown file in your Gitea user / organization profile page, simply create a repository named `.profile` and add a new file called `README.md`.
```
Comment (silverwind) [inline docs/content/usage/profile-readme.en-us.md]:
```suggestion
To display a Markdown file in your Gitea user or organization profile page, create a repository named `.profile` and add a new file named `README.md` to it.
```

File: templates/org/home.tmpl [hunk 5]
```diff
@@ -38,6 +38,10 @@
 	<div class="ui container">
 		<div class="ui mobile reversed stackable grid">
 			<div class="ui {{if .ShowMemberAndTeamTab}}eleven wide{{end}} column">
+				{{if .ProfileReadme}}
+					<div id="readme_profile" class="markup">{{.ProfileReadme | Str2html}}</div>
+				{{end}}
+
```
Comment (silverwind) [inline templates/org/home.tmpl]:
```suggestion
```

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Is it really a good design to put the "readme" there?

If the "readme" content is short, then the "org descrption" does the same thing, is the "readme" content just redundancy?

If the "readme" content is long, then it takes a lot of space of the "org home" page, then the "search input" and the "list" position/layout is quite strange.


Indeed, the "org home page"  is not the same as "user profile page".

**Comment by KN4CK3R** [comment]
Here is an example from Github: https://github.com/react-native-elements

**Comment by silverwind** [comment]
So GitHub Readmes are in special repo `.github`, is that how it works for Gitea too? Both user and org.

**Comment by wxiaoguang** [comment]
> Here is an example from Github: https://github.com/react-native-elements

Is GitHub always right?

**Review by lafriks — APPROVED** [review]

**Pull request merged**